src/go/rdctl/pkg/autostart/autostart_darwin.go

```diff
@@ -1,9 +1,109 @@
 package autostart
 
 import (
+	"bytes"
 	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/utils"
 )
 
+const launchAgentFileTemplateContents = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>io.rancherdesktop.autostart</string>
+    <key>ProgramArguments</key>
+    <array>
+        <string>{{ .RancherDesktopPath }}</string>
```

Comment:
This doesn't seem to work by itself. It looks like a workaround can be [found here](https://www.launchd.info/#/Automatically_restart_application); that is, run it via `/usr/bin/open <appdir>` (the `-W` is unnecessary for us, as there's no need to wait before we start).

Reply:
I'll try this myself once we have a working CI build. The example launchctl script I provided in #3771 was working for me, so not sure what is different now.

Reply:
The problem is that you are just writing

```xml
    <array>
        <string>/Applications/Rancher Desktop.app</string>
    </array>

```

The working script from #3771 provided the full path to the executable, not just the directory:

```xml
    <array>
        <string>/Applications/Rancher Desktop.app/Contents/MacOS/Rancher Desktop</string>
    </array>
```

An alternative is to use `/usr/bin/open -a`:

```xml
    <array>
        <string>/usr/bin/open</string>
        <string>-a</string>
        <string>/Applications/Rancher Desktop.app</string>
    </array>
```

Both work fine. Maybe the latter is preferable because it doesn't need any information about the filenames inside the app directory.